src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
-
+        if (threshold() == size) {
+            resize();
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        linkToBucket(table, newNode);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        int position = hash % table.length;
+        position = validatePosition(position, table);
```

Comment:
the index needs to be checked when it is a param that we received from outside, right now it is just a key, which can not be invalid in the same way as index, so we do not need to validate it because it was spawned by us